### PR TITLE
Add release highlights for 1.4.0

### PR DESCRIPTION
**Pull request by pebrc**
First draft for 1.4.0 release highlights. Base off of #4116 so https://github.com/elastic/cloud-on-k8s/pull/4135/commits/d9d43ee7ad9707686a96589c703929442a87c0f9 contains the relevant changes.

### CODE REVIEW

File: docs/release-notes/1.4.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+=== Enhancements
+
+* Add support for a user-defined transport CA certificate and key {pull}4053[#4053] (issue: {issue}2812[#2812])
+* Add support for DaemonSetUpdateStrategy {pull}4049[#4049] (issue: {issue}3839[#3839])
+* [Helm] Use ValidatingWebhookConfiguration v1 on supported Kubernetes versions {pull}3963[#3963] (issues: {issue}3886[#3886], {issue}3958[#3958])
+* Include Beats daemonsets in the diagnostic bundle {pull}3908[#3908]
+* Add support for webhook constraints {pull}3876[#3876] (issue: {issue}3431[#3431])
+* Remove unnecessary metrics from _nodes ES API request {pull}3860[#3860] (issue: {issue}3249[#3249])
+* Allow automatic Elasticsearch nodes discovery {pull}3837[#3837] (issues: {issue}3182[#3182], {issue}3723[#3723])
+* Enable webhook dry run on supported k8s versions {pull}3834[#3834] (issue: {issue}1233[#1233])
```
Comment (alaudazzi) [inline docs/release-notes/1.4.0.asciidoc:27]:
```suggestion
* Enable webhook dry run on supported Kubernetes versions {pull}3834[#3834] (issue: {issue}1233[#1233])
```

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+[id="{p}-140-new-and-notable"]
+=== New and notable
+
+New and notable changes in version 1.4.0 of {n}. See <<release-notes-1.4.0>> for the full list of changes.
+
+[float]
+[id="{p}-140-agent-support"]
+==== Support for Elastic Agent
+
+link:https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Elastic Agent] is a single, unified way to add monitoring for logs, metrics, and other types of data to each Kubernetes node. A single agent makes it easier and faster to deploy monitoring across your infrastructure. There is no need with agent to deploy multiple instances of Beats for different kinds of data you want to monitor. ECK 1.4.0 introduces experimental support for Elastic Agent in link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[standalone mode] as a technology preview.
```
Comment (alaudazzi) [inline docs/release-notes/highlights-1.4.0.asciidoc:14]:
```suggestion
link:https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Elastic Agent] is a single, unified way to add monitoring for logs, metrics, and other types of data to each Kubernetes node. With a single agent, it's easier and faster to deploy monitoring across your infrastructure; also there is no need to deploy multiple instances of Beats for different kinds of data you want to monitor. ECK 1.4.0 introduces experimental support for Elastic Agent in link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[standalone mode] as a technology preview.
```

File: docs/release-notes/1.4.0.asciidoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+[float]
+=== Bug fixes
+
+* Update client-go to 0.18.14 {pull}4097[#4097] (issue: {issue}4087[#4087])
+* Don't duplicate user-provided Enterprise Search encryption keys when reusing them {pull}4052[#4052] (issue: {issue}4051[#4051])
+* Add a replacement mechanism to work around nil handling in ucfg {pull}4041[#4041] (issue: {issue}3718[#3718])
+* Avoid touching root filesystem in Kibana init script {pull}4023[#4023] (issue: {issue}4022[#4022])
+* Don't allow downscales if some shards are unassigned {pull}3883[#3883] (issue: {issue}3867[#3867])
+* Include secret token in APM Server config checksum {pull}3858[#3858] (issue: {issue}523[#523])
+* Do not set the default elasticsearch-data volume if claims are provided in the ES manifest {pull}3806[#3806] (issue: {issue}2574[#2574])
```
Comment (alaudazzi) [inline docs/release-notes/1.4.0.asciidoc:40]:
```suggestion
* Do not set the default elasticsearch-data volume if claims are provided in the Elasticsearch manifest {pull}3806[#3806] (issue: {issue}2574[#2574])
```

**Review by alaudazzi — APPROVED** [review]
Left a few editing suggestions. Otherwise LGTM.

File: docs/release-notes/1.4.0.asciidoc (hunk 1 above)
Comment (pebrc) [reply docs/release-notes/1.4.0.asciidoc:27]: I will address this in #4116 (I should have mentioned that this PR is based off of #4116, sorry)

File: docs/release-notes/1.4.0.asciidoc (hunk 3 above)
Comment (pebrc) [reply docs/release-notes/1.4.0.asciidoc:40]: Same as above will address in #4116

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+[id="{p}-140-new-and-notable"]
+=== New and notable
+
+New and notable changes in version 1.4.0 of {n}. See <<release-notes-1.4.0>> for the full list of changes.
+
+[float]
+[id="{p}-140-agent-support"]
+==== Support for Elastic Agent
+
+link:https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Elastic Agent] is a single, unified way to add monitoring for logs, metrics, and other types of data to each Kubernetes node. With a single agent, it's easier and faster to deploy monitoring across your infrastructure; also there is no need to deploy multiple instances of Beats for different kinds of data you want to monitor. ECK 1.4.0 introduces experimental support for Elastic Agent in link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[standalone mode] as a technology preview.
```
Comment (charith-elastic) [inline docs/release-notes/highlights-1.4.0.asciidoc:14]: I know the description comes from the product page. It just feels like a weird way of explaining things. Feel free to ignore my suggestion though.
```suggestion
link:https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Elastic Agent] provides a unified way to monitor logs, metrics, and other types of data from your Kubernetes infrastructure quickly and easily.  A single agent deployment can be used to replace multiple Beats deployments that were previously necessary to collect the different types of data you want to monitor. ECK 1.4.0 introduces experimental support for Elastic Agent in link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[standalone mode] as a technology preview.
```

**Review by charith-elastic — APPROVED** [review]

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+[id="{p}-140-new-and-notable"]
+=== New and notable
+
+New and notable changes in version 1.4.0 of {n}. See <<release-notes-1.4.0>> for the full list of changes.
+
+[float]
+[id="{p}-140-agent-support"]
+==== Support for Elastic Agent
+
+link:https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Elastic Agent] provides a unified way to monitor logs, metrics, and other types of data from your Kubernetes infrastructure quickly and easily.  A single agent deployment can be used to replace multiple Beats deployments that were previously necessary to collect the different types of data you want to monitor. ECK 1.4.0 introduces experimental support for Elastic Agent in link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[standalone mode] as a technology preview.
```
Comment (david-kow) [inline docs/release-notes/highlights-1.4.0.asciidoc:14]: To be consistent: 

```suggestion
link:https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Elastic Agent] provides a unified way to monitor logs, metrics, and other types of data from your Kubernetes infrastructure quickly and easily. A single Elastic Agent deployment can be used to replace multiple Beats deployments that were previously necessary to collect the different types of data you want to monitor. ECK 1.4.0 introduces experimental support for Elastic Agent in link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[standalone mode] as a technology preview.
```

Also, shouldn't the first link be https://www.elastic.co/guide/en/fleet/current/elastic-agent-installation-configuration.html? The fleet-overview one talks mostly about Fleet and central management which is mostly not there for standalone mode.

**Review by david-kow — APPROVED** [review]
Two suggestions, LGTM otherwise.

**Pull request merged**